### DISCUSSION TITLE
Filtering Events Using Metadata Before Processing

### DISCUSSION THREAD

**Discussion by b-barry** [spatie/laravel-event-sourcing#482] [Q&A] [answered]
I'm interested in the possibility of filtering events based on metadata before they reach projections or aggregates in spatie/laravel-event-sourcing. Specifically:

Is there a built-in mechanism to filter events based on their metadata before they are processed by projections or aggregates? For example, only processing MoneyAdded events where the amount is greater than 1000.

**Comment by inmanturbo** [comment, marked as answer]
The docs explain how to implement business rules in the aggregate, before
recording the event. In projectors you type hint in the method signature
which event you want the method to handle. You are then free to do what you
want with the event and data based on any conditions or logic you require,
but at that point the event has already been recorded.
